clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1512,6 +1512,66 @@ void SetElementTypeAsReturnType(Sema *S, CallExpr *TheCall,
   TheCall->setType(ReturnType);
 }
 
+bool CheckBoolSelect(Sema *S, CallExpr *TheCall) {
```

Comment:
nit: it'd be good for these helpers that are only used from this cpp file to be marked as static.  (I see that the ones above are not marked as static...I'd be tempted to fix them as well in another PR).